include/hal.h

```diff
@@ -46,8 +46,8 @@ void hal_deinit();
 #endif
 
 void hal_init(void);
-int hal_flash_write(uint32_t address, const uint8_t *data, int len);
-int hal_flash_erase(uint32_t address, int len);
+int hal_flash_write(uintptr_t address, const uint8_t *data, int len);
```

Comment:
@danielinux , this change of mine seems like it could break some platforms. Perhaps I could just add a HAL build option to support 64-bit addresses?